binary_heap.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+
+class BinaryHeap(object):
+    """Implements a binary max-heap data structure"""
+
+    def __init__(self, an_iter=None):
+        '''Initializes the heap,
+        optionally populates it with the given iterable
+        '''
+
+        self._list = []
+
+        if an_iter:
+            for val in an_iter:
+                self.push(val)
+
+    def push(self, val):
+        '''Pushes the value into the heap,
+         adjusting the ordering to maintain the heap
```

Comment:
Try to fit all the vital stuff in a docstring into one line.  Second lines should be for additional info and should be separated from the first by one empty line.